components/profile_popover/profile_popover.tsx

```diff
@@ -16,263 +13,270 @@ import {browserHistory} from 'utils/browser_history';
 import * as GlobalActions from 'actions/global_actions';
 import Constants, {ModalIdentifiers, UserStatuses} from 'utils/constants';
 import {t} from 'utils/i18n';
-import {intlShape} from 'utils/react_intl';
 import * as Utils from 'utils/utils.jsx';
 import Pluggable from 'plugins/pluggable';
 import AddUserToChannelModal from 'components/add_user_to_channel_modal';
 import LocalizedIcon from 'components/localized_icon';
 import ToggleModalButtonRedux from 'components/toggle_modal_button_redux';
 import Avatar from 'components/widgets/users/avatar';
 import Popover from 'components/widgets/popover';
-import SharedUserIndicator from 'components/shared_user_indicator.tsx';
+import SharedUserIndicator from 'components/shared_user_indicator';
 import CustomStatusEmoji from 'components/custom_status/custom_status_emoji';
 import CustomStatusModal from 'components/custom_status/custom_status_modal';
 import CustomStatusText from 'components/custom_status/custom_status_text';
+import {UserCustomStatus, UserProfile, UserTimezone} from 'mattermost-redux/types/users';
+import {Dictionary} from 'mattermost-redux/types/utilities';
 
 import './profile_popover.scss';
 
+interface ProfilePopoverProps extends Omit<React.ComponentProps<typeof Popover>, 'id'>{
+
+    /**
+     * Source URL from the image to display in the popover
+     */
+    src: string;
+
+    /**
+     * Source URL from the image that should override default image
+     */
+    overwriteIcon?: React.ReactNode;
```

Comment:
Shouldn't this also be a `string`?

Reply:
<img width="837" alt="Screenshot 2021-06-07 at 8 55 33 PM" src="https://user-images.githubusercontent.com/16203333/121044478-2b18bb00-c7d3-11eb-9666-e56f369b1335.png">
That's why I added them as ReactNode but yeah it does make sense to keep them as a string

Reply:
Done

Reply:
Conflict with other components if we put there type as string

Reply:
@hmhealey @devinbinnie please suggest what should be the course of action here

Reply:
My main concern is that in one Component we are expecting string and in one React.ReactNode can use the Union type of React.ReactNode and String and then cast them accordingly.

Reply:
Well, `overwriteIcon` should always be a `string` since it's used as the `src` attribute for an `img` tag. As for `overwriteName`, I think it can stay as `ReactNode` because `UserProfile` expects that you can pass any arbitrary HTML into it as an override. So we can probably also change the code referenced [here](https://github.com/mattermost/mattermost-webapp/pull/8216#discussion_r646703351) to:
`title = this.props.overwriteName`

Reply:
Type of overwriteIcon is ReactNode in UserProfile is ReactNode, We are using UserProfile at a couple more place so changing that to string doesn't feel like the right choice to me without assessing the risk of that.
<img width="976" alt="Screenshot 2021-06-09 at 2 49 24 PM" src="https://user-images.githubusercontent.com/16203333/121329313-ba88b000-c932-11eb-9520-939e270a7f0b.png">
<img width="370" alt="Screenshot 2021-06-09 at 2 49 05 PM" src="https://user-images.githubusercontent.com/16203333/121329326-bf4d6400-c932-11eb-8cd7-6216e4db66c9.png">

One solution comes to my mind as overwriteIcon is optional we can do something like this in UserProfile
```
<ProfilePopover
 className='user-profile-popover'
 userId={userId}
channelId={channelId}
src={profileImg}
isBusy={isBusy}
hide={this.hideProfilePopover}
hideStatus={hideStatus}
isRHS={isRHS}
hasMention={hasMention}
overwriteName={overwriteName}
overwriteIcon={typeof overwriteIcon === 'string' ? overwriteIcon : ''}
 />
```
I think after this we don't need to change the type of overwriteIcon inside UserProfile and we can keep the type safety of typescript as well.

Reply:
The root of the issue here is that `overwriteIcon` is incorrectly defined in `UserProfile`, it should always be a `string`. The only usage of `overwriteIcon` is passed into `ProfilePopover` (which does specify that `overwriteIcon` is a string) which is then passed into `Avatar` (also expecting a `string`) which passes it into the `img` tag. I don't see a lot of risk here in changing that definition since it's wrong. I know it increases the scope a bit here but it's definitely the most correct thing to do. We shouldn't need to be doing runtime type checking for stuff like this.

Reply:
Great thanks will change it there as well.